docs/content/users/configuration/config_yaml.md

```diff
@@ -1,75 +1,74 @@
+| type                             | Project type                                                                              | Can be `php`, `drupal6`, `drupal7`, `drupal8`, `backdrop`, `typo3`, or `wordpress`. The `php` type doesn’t attempt CMS configuration or settings file management and can work with any project.                                                                                                                                                                                                                                             |
+| docroot                          | Relative path to docroot containing `index.php` / `index.html`                            |                                                                                                                                                                                                                                                                                                                                                                                                                                             |
+| php_version                      | `5.6`, `7.0`, `7.1`, `7.2`, `7.3`, `7.4`, `8.0`, `8.1`                                    | You can only specify the major version (`7.3`), not a minor version (`7.3.2`), from those explicitly available.                                                                                                                                                                                                                                                                                                                             |
+| webimage                         | Docker image to use for web server                                                        | It’s unusual and not recommended to change the default, but the web image can be overridden with a correctly-crafted image, probably based on [drud/ddev-webserver](https://github.com/drud/ddev/tree/master/containers/ddev-webserver).                                                                                                                                                                                                    |
+| database                         | Database `type` and `version` to be used                                                  | Defaults to MariaDB 10.3.<br><br>MariaDB 5.5–10.7, MySQL 5.5–8.0, and PostgreSQL 9–14 are available. See [Database Server Types](../extend/database_types.md) for details and caveats.<br><br>DDEV v1.18 and earlier used `mariadb_version` and `mysql_version`, which are now automatically converted to the `database` format.                                                                                                            |
```

Comment:
Wait, can you actually use a single string like `mariadb:5.5` instead of explicit, separate `type: mariadb` + `version: 5.5`? I specifically changed that format because I thought `type` and `version` were required keys and `mariadb:5.5` suggested a format you can’t actually use.

Reply:
Oh shoot, this is the config_yaml, not the args to ddev config. You're right. But we're going to have to do it better. The "type" key is lower-case, `mariadb` not `MariaDB`.  I see you've got some work done on using HTML to format this right, maybe we can give more examples.

Reply:
I generally tried to cut characters from this table and make it more consistent since its primary aim is to be a comprehensive reference. The page it links to in this case immediately does a perfect job of illustrating how to specify a database engine and version—no need to try and do that here as well since explanation gets in the way.

It’d be nice to find a format that can neatly accommodate DDEV configuration options _period_, regardless of whether they’re set via YAML or command options—that’s just beyond the scope of immediate cleanup.

Reply:
This stuff should be able to be generated by the build, instead of being maintained this way.

Reply:
That’s on my list too! Not necessarily for this page, but an exhaustive list somewhere of the latest DDEV version’s included software and individual versions. Automatically updated for everyone’s joy.